rules/opentitan_test.bzl

```diff
@@ -452,3 +454,138 @@ def opentitan_functest(
             # For more see https://bazel.build/reference/be/general#test_suite.tags
         ],
     )
+
+def _opentitan_sram_fpga_cw310_test(ctx):
+    test_script = '''#!/usr/bin/env bash
+set -e
+
+COLOR_RED='\\x1b[0;31m'
+COLOR_GREEN='\\x1b[0;32m'
+COLOR_PURPLE='\\x1b[0;35m'
+
+function prefix_lines() {{
+    LABEL="$1"
+    COLOR="$2"
+    COLOR_RESET='\\x1b[m'
+    sed -Eu "s/(.*)/$COLOR[$LABEL]$COLOR_RESET \\1/"
+}}
+
+set -x
+
+./opentitantool load-bitstream --rom-kind=TestRom test_rom.bit
+
+(openocd -f /usr/share/openocd/scripts/interface/ftdi/olimex-arm-usb-tiny-h.cfg \\
+    -c "adapter speed 500; transport select jtag; reset_config trst_and_srst" \\
+    -f {openocd_earlgrey_config} 2>&1 \\
+    | prefix_lines OPENOCD "$COLOR_PURPLE") &
+OPENOCD_PID=$!
+
+(/tools/riscv/bin/riscv32-unknown-elf-gdb --command=gdb_script 2>&1 \\
+    | prefix_lines GDB "$COLOR_GREEN") &
+
+./opentitantool console \\
+    --timeout 15s \\
+    --exit-success='{exit_success_pattern}' \\
+    | prefix_lines CONSOLE "$COLOR_RED"
+
+# Send TERM signal to OpenOCD and wait for all background jobs to complete. Note
+# that GDB will exit naturally at the end of its script.
+kill $OPENOCD_PID
+wait
+    '''.format(
+        openocd_earlgrey_config = ctx.file.openocd_earlgrey_config.path,
+        exit_success_pattern = ctx.attr.exit_success_pattern,
+    )
+
+    # Write a GDB script to disk and load it with GDB's `--command` argument.
+    # This enables us to separate lines with whitespace, whereas if we piped the
+    # string into GDB's stdin, each newline would cause it to repeat the
+    # previous command.
+    gdb_script = """target extended-remote :3333
```

Comment:
Make gdb script a parameter.

Reply:
@sriyerg It sounded like you're probably not going to go the GDB route. If that's the case, I'd rather not over-generalize this by parameterizing the GDB script.

Reply:
Yeah, sure. I can always come back to this commit and read the text if we want to mimic the gdb path in our DV environment.

Reply:
Yes, this should be a parameter.